core/app/models/spree/variant.rb

```diff
@@ -65,7 +65,7 @@ class Variant < Spree::Base
     self.whitelisted_ransackable_attributes = %w[weight sku]
 
     def self.active(currency = nil)
-      not_discontinued.joins(:prices).where(deleted_at: nil).where('spree_prices.currency' => currency || Spree::Config[:currency]).where('spree_prices.amount IS NOT NULL')
+      not_discontinued.joins(:prices).where(deleted_at: nil).where('spree_prices.currency' => currency || Spree::Config[:currency]).where('spree_prices.amount IS NOT NULL').uniq
```

Comment:
@priyank-gupta - Please move it in scope.

Also, it will be good if we can move the required scopes in Price model and use merge here.